### PR TITLE
corecfg:  support setting proxy.store if there's a matching store assertion

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
based on #4106

File: corecfg/proxy.go [hunk 1]
```diff
@@ -71,3 +74,25 @@ func handleProxyConfiguration(tr Conf) error {
...
 	return nil
 }
+
+func handleProxyStore(tr Conf) error {
+	proxyStore, err := coreCfg(tr, "proxy.store")
+	if err != nil {
+		return err
+	}
+
+	if proxyStore != "" {
```
Comment (mvo5) [inline corecfg/proxy.go]: (nitpick) I personally would make this `if proxyStore == nil { return nil }`

Comment (pedronis) [reply corecfg/proxy.go]: how can I do that? proxyStore is a string, and "" is also the only way to unset config atm

Comment (mvo5) [reply corecfg/proxy.go]: Ups, sorry, silly me. I should have written: `if proxyStore != "" { return nil }`, i.e. just to return early not to change any logic.

Comment (pedronis) [reply corecfg/proxy.go]: ah, the nil confused me

**Pull request merged**